tests/test_pipeline/test_hierarchical_pipeline.py

```diff
@@ -0,0 +1,119 @@
+from etna.reconciliation import TopDownReconciliator
+
+
+@pytest.fixture
+def market_level_constant_hierarchical_ts(hierarchical_structure):
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02", "2000-01-03"] * 2,
+            "segment": ["X"] * 3 + ["Y"] * 3,
+            "target": [2, 2, 2] + [2, 2, 2],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    ts = TSDataset(df=df, freq="D", hierarchical_structure=hierarchical_structure)
+    return ts
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="a", source_level="b", period=1, method="AHP"),
+        BottomUpReconciliator(target_level="a", source_level="b"),
+    ),
+)
+def test_init_pass(reconciliator):
+    model = MovingAverageModel(window=1)
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
+    assert isinstance(pipeline.reconciliator, type(reconciliator))
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="market", source_level="total", period=1, method="AHP"),
+        BottomUpReconciliator(target_level="total", source_level="market"),
+    ),
+)
+def test_fit_mapping_matrix(market_level_simple_hierarchical_ts, reconciliator):
+    model = MovingAverageModel(window=1)
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
+    pipeline.fit(market_level_simple_hierarchical_ts)
+    assert pipeline.reconciliator.mapping_matrix is not None
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="market", source_level="total", period=1, method="AHP"),
+        BottomUpReconciliator(target_level="total", source_level="market"),
+    ),
+)
+def test_fit_dataset_level(market_level_simple_hierarchical_ts, reconciliator):
+    model = MovingAverageModel(window=1)
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
+    pipeline.fit(market_level_simple_hierarchical_ts)
+    assert pipeline.ts.current_df_level == reconciliator.source_level
+
+
+def test_fit_no_hierarchy(simple_no_hierarchy_ts):
+    model = MovingAverageModel(window=1)
+    reconciliator = BottomUpReconciliator(target_level="total", source_level="market")
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
+    with pytest.raises(ValueError, match="The method can be applied only to instances with a hierarchy!"):
+        pipeline.fit(simple_no_hierarchy_ts)
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="market", source_level="total", period=1, method="AHP"),
+        BottomUpReconciliator(target_level="total", source_level="market"),
+    ),
+)
+def test_raw_forecast_level(market_level_simple_hierarchical_ts, reconciliator):
```

Comment:
May you also add tests for `forecast` and `raw_forecast` for the correct values in target. Also add regressors in fixture and transforms in the pipeline. Check that there is no column duplication after `forecast` for exog columns.